x/ccv/provider/keeper/keeper.go

```diff
@@ -392,6 +392,52 @@ func (k Keeper) GetUnbondingOpsFromIndex(ctx sdk.Context, chainID string, valset
 	return entries, true
 }
 
+// GetMaturedUnbondingOps returns the list of matured unbonding operation ids
+func (k Keeper) GetMaturedUnbondingOps(ctx sdk.Context) (ids []uint64, err error) {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.MaturedUnbondingOpsKey())
+	if bz == nil {
+		return nil, nil
+	}
+	err = json.Unmarshal(bz, &ids)
+	if err != nil {
+		return nil, err
+	}
+	return ids, nil
+}
+
+// AppendMaturedUnbondingOps adds a list of ids to the list of matured unbonding operation ids
+func (k Keeper) AppendMaturedUnbondingOps(ctx sdk.Context, ids []uint64) error {
+	if len(ids) == 0 {
+		return nil
+	}
+	existingIds, err := k.GetMaturedUnbondingOps(ctx)
+	if err != nil {
+		return err
+	}
+	// append works also on a nil list
+	existingIds = append(existingIds, ids...)
+
+	store := ctx.KVStore(k.storeKey)
+	bz, err := json.Marshal(existingIds)
```

Comment:
nit: more standard to marshal/unmarshal in stores with protobuf and wrap a list in a proto type

Reply:
@AdityaSripal Not sure how to do that. Do you have an example (from cosmos-sdk or ibc-go)?

Reply:
https://github.com/cosmos/cosmos-sdk/blob/76bbcd2ce3/proto/cosmos/distribution/v1beta1/distribution.proto#L93

https://github.com/cosmos/cosmos-sdk/blob/76bbcd2ce38fd0b56d999da41ba9ea630f5ef1b4/x/distribution/keeper/store.go#L47

Not a huge deal, just a note

Reply:
Yeah I do 👍